temporal-testing/src/main/java/io/temporal/internal/testservice/StateUtils.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.testservice;
+
+public class StateUtils {
+  /** @return true if the workflow was completed not by workflow task event */
+  public static boolean isWorkflowExecutionTerminatedExternally(StateMachines.State state) {
```

Comment:
I would rename the function to isWorkflowExecutionForcefullyCompleted. The current name sounds weird for timed out case.

Reply:
👍 I was thinking about naming for some time. Will go with your version.